network-protocols.md

```diff
@@ -307,30 +307,41 @@ func OnDataReceived(prop StorageDealProposal) {
   }
   
   // TODO: is TranslatedRef actually needed? How does it tie in?
-  pieceConfirmation := SectorBuilder.PackData(prop.PieceRef, prop.SerializationMode) 
-  // TODO: we can't get a *full* piece confirmation until the sector is actually full.
-  // but we can return something useful here... can't we? Whatever it is should likely
-  // be related to the prop.TranslatedRef in some (proveable) way.
+  SectorBuilder.AddPiece(prop.PieceRef, prop.SerializationMode) 
+}
+```
+
+```go
+func OnSectorPacked(prop StorageDealProposal, pieceConf PieceCommitment) {
   resp := StorageDealResponse{
     State: Staged,
     Proposal: prop.Cid(),
-    PieceConfirmation: pieceConfirmation,
+    PieceConfirmation: pieceConf,
   }
+  
   miner.Sign(resp)
   miner.SetDealState(resp)
 }
 ```
 
+Once the deal makes it to this state, the client should be able to query and get the `PieceConfirmation` that they need to verify that the miner is indeed storing their data.
```

Comment:
I'm Lovin' It.

Reply:
@whyrusleeping Although this should be obvious, I want to make explicit that if we do this in the simplest way, `PieceConfirmation`s will take some time to generate because we also have to generate `commD`. So the likely flow is:
 - sector is completely packed.
 - `commD` and `PieceConfirmation`s are generated.
 - `OnSectorPacked` is called.
 -  Sealing beings/continues.

We could also incrementally generate `commD` as pieces are packed, but that's probably not the effort, at least up-front. Like I said, should be obvious, but there's likely to be a delay between when the sector is completely packed and when `OnSectorPacked` can be called. I can't think of a reason why this would matter.